lib/Dialect/FIRRTL/Transforms/InferResets.cpp

```diff
@@ -55,50 +55,67 @@ using namespace firrtl;
 // Utilities
 //===----------------------------------------------------------------------===//
 
+/// Return the name and parent module of a reset. The reset value must either be
+/// a module port or a wire/node operation.
+static std::pair<StringAttr, FModuleOp> getResetNameAndModule(Value reset) {
+  if (auto arg = dyn_cast<BlockArgument>(reset)) {
+    auto module = cast<FModuleOp>(arg.getParentRegion()->getParentOp());
+    return {module.getPortNameAttr(arg.getArgNumber()), module};
+  }
+  auto *op = reset.getDefiningOp();
+  return {op->getAttrOfType<StringAttr>("name"),
+          op->getParentOfType<FModuleOp>()};
+}
+
+/// Return the name of a reset. The reset value must either be a module port or
+/// a wire/node operation.
+static StringAttr getResetName(Value reset) {
+  return getResetNameAndModule(reset).first;
+}
+
 namespace {
 /// A reset domain.
 struct ResetDomain {
   /// Whether this is the root of the reset domain.
   bool isTop = false;
+
   /// The reset signal for this domain. A null value indicates that this domain
   /// explicitly has no reset.
-  Value reset;
-
-  // Implementation details for this domain.
-  Value existingValue;
+  Value rootReset;
```

Comment:
What is the current interaction between `isTop` and `rootReset`? Is the former now unnecessary?

Reply:
At this point, `rootReset` is really only there for error messages. I do think there is some redundant information in this struct, but I am not sure how to factor it.  In the end I decided to just keep it simple, but maybe we can tackle this more in the future? For example, I am not sure why `buildDomains` and `determineImpl` are separate phases.  If we squash them in to one, we don't have to hold on to so much information, and could delete the `isTop` field.  As well, something like the `existingPort` could be determined by casting the local reset to a block arg and asking for the port number.  There are probably some other things we could do.